File: Algoexpert/Minimum Waiting Time/minimum_waiting_time.py
def minimumWaitingTime(queries):

    queries.sort()
    time_taken = 0
    for i in range(len(queries)-1):
        time_taken += sum(queries[0:i+1])
        
    return time_taken

def minimumWaitingTime_enumeration(queries):
    
    queries.sort()

    time_taken = 0
    for i, duration in enumerate(queries):
        queries_left = len(queries) - (i + 1)
        time_taken += duration * queries_left

    return time_taken

def main():
    
    array = [3, 2, 1, 2, 6]
    
    print(minimumWaitingTime(array))
    print(minimumWaitingTime_enumeration(array))
    
if __name__ == "__main__":
    main()

File: Algoexpert/Minimum Waiting Time/cpp/minimum_waiting_time.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
    public:
    int minimumWaitingTime(vector<int>& queries) {
        
        sort(queries.begin(), queries.end());
        int totalWaitingTime = 0;

        for(size_t i = 0; i < queries.size(); i++) {
            int queriesLeft = queries.size() - (i+1);
            totalWaitingTime += queriesLeft * queries[i];
        }

        return totalWaitingTime;
    }
    
};

int main() {

    vector<int> queries = {3, 2, 1, 2, 6};

    Solution solution;

    cout << solution.minimumWaitingTime(queries) << endl;

    return 0;
}